tests/api2/test_openssl.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+
+from middlewared.test.integration.utils import call, ssh
+from auto_config import ha
+
+retry = 5
+fips_version = "3.0.9"
+
+
+# Sometimes this test fails because the testing environment has broken failover (randomly. Fun transient error. Reports a failed heartbeat).
+@pytest.mark.flaky(reruns=retry, reruns_delay=5)
+@pytest.mark.skipif(not ha, reason='Test only valid for HA')
+def test_fips_version():
+    # The reason we have a set of commands in a payload is because of some annoying FIPS technicalities.
+    # Basically, when FIPS is enabled, we can't use SSH because the SSH key used by root isn't using a FIPS provided algorithm. (this might need to be investigated further)
+    # To allow testing, we write our FIPS information to a file during this phase, and then go disable FIPS to get SSH back all in one joint command.
+    payload = """midclt call --job system.security.update '{"enable_fips": true}' && openssl list -providers > /root/osslproviders && midclt call system.reboot.info >> /root/osslproviders && midclt call --job system.security.update '{"enable_fips": false}'"""
```

Comment:
I am confused, why this is being done `openssl list -providers > /root/osslproviders` because the same file is overwritten in the next step of the chain ?

Reply:
No the next command appends to the file (>>) as opposed to overwrite (>)

Reply:
Oops, missed that